vpr/src/util/vpr_utils.cpp

```diff
@@ -606,9 +633,27 @@ std::tuple<ClusterNetId, int, int> find_pb_route_clb_input_net_pin(ClusterBlockI
     //curr_pb_pin should be a top-level CLB input
     ClusterNetId clb_net_idx = cluster_ctx.clb_nlist.block_net(clb, curr_pb_pin_id);
     int clb_net_pin_idx = cluster_ctx.clb_nlist.block_pin_net_index(clb, curr_pb_pin_id);
-    VTR_ASSERT(clb_net_idx != ClusterNetId::INVALID());
+
+    /* The net could be remapped, we should verify the remapped location */
+    auto remapped_clb = cluster_ctx.post_routing_clb_pin_nets.find(clb);
+    if (remapped_clb != cluster_ctx.post_routing_clb_pin_nets.end()) {
+        auto remapped_result = remapped_clb->second.find(curr_pb_pin_id);
+        if ((remapped_result != remapped_clb->second.end())
+            && (remapped_result->second != clb_net_idx)) {
+            clb_net_idx = remapped_result->second;
+            VTR_ASSERT(clb_net_idx);
+            clb_net_pin_idx = cluster_ctx.clb_nlist.block_pin_net_index(clb, cluster_ctx.pre_routing_net_pin_mapping.at(clb).at(curr_pb_pin_id));
+        }
+    }
     VTR_ASSERT(clb_net_pin_idx >= 0);
 
+    /* TODO: REMOVE AFTER DEBUGGING IS FINISHED
```

Comment:
Add an issue to remove debug code, mark the PR as WIP, or just stash it out of the PR.